### ISSUE TITLE
attribute-hyphenation should not be applied to dynamic arguments

### ISSUE THREAD

**Issue by msamsel** [vuejs/eslint-plugin-vue#1106]
**Tell us about your environment**	
* **ESLint version:**	`^5.16.0`
* **eslint-plugin-vue version:**	`^5.0.0`
* **Node version:**	`v12.14.1`

**Please show your full configuration:**

```js
module.exports = {
  root: true,

  env: {
    node: true,
  },

  extends: [
    'plugin:vue/essential',
    'plugin:vue/recommended',
    '@vue/airbnb',
  ],

  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'off' : 'warn',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-shadow': ['error', { allow: ['state', 'getters'] }],
    'no-cycle': 'off',
    camelcase: 'off',
    'no-underscore-dangle': ['error', { allow: ['_acl'] }],
    'vue/no-v-html': 'off',
  },

  parserOptions: {
    parser: 'babel-eslint',
  },

  overrides: [
    {
      files: [
        '**/tests/e2e/**/*.test.js',
        '**/tests/integration/**/*.test.js',
        '**/tests/unit/**/*.test.js',
      ],
      env: {
        jest: true,
      },
    },
  ],
};
```

**What did you do?**	

```vue
<template>
  <component
    :is="componentName"
    :[customProp]="true"
  />
</template>

<script>
export default {
  data() {
    return {
      componentName: 'Foo',
    };
  },

  computed: {
    customProp() {
      return 'foo';
    },
  },
};
</script>
```

**What did you expect to happen?**
Linter should not apply `vue/attribute-hyphenation` rule for [dynamic arguments](https://vuejs.org/v2/guide/syntax.html#Dynamic-Arguments)

**What actually happened?**
```
yarn eslint ./src/components/settings/foo.vue               ✔  11:25:22 
yarn run v1.21.1
$ /Users/msamsel/…/node_modules/.bin/eslint ./src/components/settings/foo.vue

/Users/msamsel/…/src/components/settings/foo.vue
  4:5  warning  Attribute ':[customProp]' must be hyphenated  vue/attribute-hyphenation

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

✨  Done in 0.93s.
```

**Comment by ota-meshi** [comment]
Please use v6.x.
v5.x does not parse dynamic arguments.
